changelog/16.changed.md

```diff
@@ -0,0 +1 @@
+Functions that previously returned `.to_dict()` responses now return `sanitize_for_serialization()` responses. The structure is similar but may have slight differences in nested object handling.
```

Comment:
suggestion: I don't think we should expect users to know about Kubernetes Python API client internals. Maybe phrase it like this:

```suggestion
API response transformation has been altered. All functions that returned resource dictionaries with `snake_case` keys (such as `show_deployment`) now return keys in `camelCase` instead, as found in YAML manifest files. The structure is similar, but may have slight differences in nested object handling. Dates and times might be reported differently.
```